### ISSUE TITLE
Fatal playback failure due to phantom media segment request

### ISSUE THREAD

**Issue by romsil** [google/shaka-player#3717]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**What version of Shaka Player are you using?**
3.2.0

**Can you reproduce the issue with our latest release version?**
Yes

**Can you reproduce the issue with the latest code from `master`?**
Yes

**Are you using the demo app or your own custom app?**
Custom app.

**If custom app, can you reproduce the issue using our demo app?**
I believe so. It is difficult to reproduce the issue since it only happens on certain ads on our player.

**What browser and OS are you using?**
Chrome, macOS

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**
N/A

**What are the manifest and license server URIs?**
Our manifest URIs are short-lived, so I am pasting the period declaration instead:
```
<Period duration="PT30.720000S" id="a-2">
      <AssetIdentifier schemeIdUri="urn:com:uplynk:ad-asset-id" value="d4bfa4522c044a968fc4f2a0d944340f" />
      <AdaptationSet audioSamplingRate="48000" contentType="audio" group="1" lang="en" mimeType="audio/mp4" segmentAlignment="true" startWithSAP="1">
         <Role schemeIdUri="urn:mpeg:dash:role:2011" value="main" />
         <Representation bandwidth="127878" codecs="mp4a.40.5" id="v0">
            <AudioChannelConfiguration schemeIdUri="urn:mpeg:dash:23003:3:audio_channel_configuration:2011" value="2" />
            <BaseURL>https://RedactedCDNUrl/slices/d4b/4f59ea65df184a2580ad8fc80d3eaa32/d4bfa4522c044a968fc4f2a0d944340f/</BaseURL>
            <SegmentTemplate duration="184320" initialization="https://RedactedCDNUrl/slices/d4b/4f59ea65df184a2580ad8fc80d3eaa32/d4bfa4522c044a968fc4f2a0d944340f/TK_init.mp4" media="TK_$Number%08d$.m4f" startNumber="0" timescale="90000" />
         </Representation>
      </AdaptationSet>
```
See notes below on our investigation results.

**What configuration are you using?  What is the output of `player.getConfiguration()`?**
Mostly default. See attached screenshot for `player.getConfiguration()` output:

![ShakaPlayerConfigs](https://user-images.githubusercontent.com/20781478/138963184-3d00fb72-141f-4930-8d72-2c1887878dfe.png)

**What did you do?**
I just let the asset play until it reaches the ad period.

**What did you expect to happen?**
Playback will continue with no fatal error.

**What actually happened?**
Shaka makes a network request to media segment `TK_00000015.m4f`.
Based on the period declaration for the audio Adaptation Set, the media segments from `TK_00000000.m4f` to `TK_00000014.m4f` should have been enough. See attached screenshot of my computation.

![AdPeriodMediaSegmentList](https://user-images.githubusercontent.com/20781478/138963620-938ea906-c23c-4655-9a73-0ae96161d4de.png)

We investigated the issue and we think this is caused by small floating points being ceiled in Shaka. We have patched Shaka below to fix the issue.
```
--- a/lib/dash/segment_template.js
+++ b/lib/dash/segment_template.js
@@ -345,7 +345,7 @@ shaka.dash.SegmentTemplate = class {
       // In period-relative positions (0-based).
       const availablePeriodPositions = [
         Math.ceil(availablePeriodTimes[0] / segmentDuration),
-        Math.ceil(availablePeriodTimes[1] / segmentDuration) - 1,
+        Math.ceil((availablePeriodTimes[1] / segmentDuration).toFixed(10)) - 1,
       ];
```
It's a small one-line fix but it's causing playback failure.

**Comment by boredom2** [comment]
Hi there,

as Shaka v3.3 is now initially released, this Problem here would be our biggest issue.
As far as I can see, this has not been included in the 3.3 Release, right?
When can we expect this to be merged into a Release?

Thank you very much and btw - Happy New Year :)

**Comment by joeyparrish** [comment]
This issue has not been fixed yet.  The fix in PR #3721 was not merged due to some debate that was never resolved.  I'll give it another look now.
